torch/distributed/tensor/parallel/ddp.py

```diff
@@ -0,0 +1,90 @@
+from typing import Any, List, Tuple
+
+import torch.nn as nn
+from torch.distributed.tensor.parallel.fsdp import _flatten_tensor, _unflatten_tensor
+
+__all__ = ["pre_dp_module_transform"]
+
+
+def _get_submodule_n_params(module: nn.Module, path: str):
+    """
+    Get submodule and the direct path of parameter from the module
+    """
+    if "." in path:
+        path_list = path.split(".")
+        parent_module_path = ".".join(path_list[:-1])
+        module = module.get_submodule(parent_module_path)
+        path = path_list[-1]
+    return module, path
+
+
+def _update_module_param(param_list: List[Tuple[nn.Module, str, nn.Parameter]]):
+    """
+    Update parameters within the module
+    """
+    for item in param_list:
+        parent_module, module_path, t = item
+        assert hasattr(parent_module, module_path)
+        delattr(parent_module, module_path)
+        setattr(parent_module, module_path, t)
+
+
+def _reconstruct_dtensor(module: nn.Module, _input: Any):
+    """
+    Recontruct DTensor parameters from local tensors
+    """
+    param_list = []
+    for name, t in module.named_parameters():
```

Comment:
can be done in follow up PRs: I am wondering if we should memorize those dtensor parameter paths as a dict, so that we can do localize/reconstruct dtensor faster instead of looping through all module parameters, because sometimes # of parameters can be quite large

Reply:
yeah, I agree, there are some spaces for optimizations. Will think more on that.